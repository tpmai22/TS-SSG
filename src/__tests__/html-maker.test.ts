const ReactTestRenderer = require('react-test-renderer');
const { processingFile, processMarkdown } = require('../html-maker');

describe('testing HTML generator', () => {
  it('should return blank with invalid file extension', () => {
    const fileName = 'test.dat';
    expect(processingFile(fileName)).toEqual('');
  });

  it('should return correct markdown for mock content', () => {
    const mockData = '## Testing testing';
    expect(processMarkdown(mockData)).toMatch('<h2>Testing testing</h2>');
  });
});

it('processMarkdown to HTML renders correctly', () => {
  const data = `# this is a header 1 
   # this is a **header** line.
   also regular.
   ## this is a header again.
   ### heading wooooooo
   that the 2 lines "this is header line" and "`;
  let testData = processMarkdown(data);

  const tree = ReactTestRenderer.create(testData).toJSON();
  expect(tree).toMatchSnapshot();
});
