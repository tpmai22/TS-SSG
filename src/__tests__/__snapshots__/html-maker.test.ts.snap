// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`processMarkdown to HTML renders correctly 1`] = `
"<h1>this is a header 1</h1>
<h1>this is a <strong>header</strong> line.</h1>
<p>also regular.</p>
<h2>this is a header again.</h2>
<h3>heading wooooooo</h3>
<p>that the 2 lines &quot;this is header line&quot; and &quot;</p>
"
`;
